include/coap/option.h

```diff
@@ -407,4 +409,68 @@ coap_opt_const_value( const coap_opt_t *opt ) {
 
 /** @} */
 
+/**
+ * Representation of chained list of CoAP options to install.
+ *
+ * @code
+ * coap_optlist_t *optlist_head = NULL;
+ * coap_pdu_t *pdu = coap_new_pdu(session);
+ *
+ * ... other set up code ...
+ * coap_insert_optlist(&optlist, coap_new_optlist(COAP_OPTION_OBSERVE,
+ *                    COAP_OBSERVE_ESTABLISH, NULL));
+ *
+ * coap_add_optlist_pdu(pdu, *optlist_head);
+ * @endcode
+ */
+typedef struct coap_optlist_t {
+  struct coap_optlist_t *next;  /**< next entry in the optlist chain */
+  uint16_t number;              /**< the option number (no delta coding) */
+  unsigned int length;          /**< the option value length */
```

Comment:
Given that the length argument in `coap_new_optlist()` is of type `size_t`, it is weird that the field `length` here is of type `unsigned int` which may or may not be the same as `size_t`.

Reply:
Fixed in update to be consistent with size_t